persist/fs/mmap.go

```diff
@@ -41,64 +41,79 @@ type mmapOptions struct {
 	read bool
 	// write is whether to make mmap bytes ref writable
 	write bool
-	// hugePages is the mmap huge pages options
-	hugePages mmapHugePagesOptions
+	// hugeTLB is the mmap huge TLB options
+	hugeTLB mmapHugeTLBOptions
 }
 
-type mmapHugePagesOptions struct {
-	// enabled determines if using huge pages is enabled for platforms
+type mmapResult struct {
+	result  []byte
+	warning error
+}
+
+type mmapHugeTLBOptions struct {
+	// enabled determines if using the huge TLB flag is enabled for platforms
 	// that support it
 	enabled bool
 	// threshold determines if the size being mmap'd is greater or equal
-	// to this value to use or not use transparent huge pages if enabled
+	// to this value to use or not use the huge TLB flag if enabled
 	threshold int64
 }
 
-func mmapFiles(opener fileOpener, files map[string]mmapFileDesc) error {
+func mmapFiles(opener fileOpener, files map[string]mmapFileDesc) (error, error) { // nolint: golint
```

Comment:
I think we should create another struct here, return two untype errors is quite weird.  Also would rather not escape the linter if we don't have to of course.

i.e. 

```go
type mmapFilesResult struct {
  warning error
}

func mmapFiles(...) (mmapFilesResult, error) { 
  ...
}
```